src/Components/TemporaryDrawer.js

```diff
@@ -0,0 +1,305 @@
+import * as React from "react";
+import Box from "@mui/material/Box";
+import Drawer from "@mui/material/Drawer";
+import Button from "@mui/material/Button";
+import List from "@mui/material/List";
+import Divider from "@mui/material/Divider";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import ListItemText from "@mui/material/ListItemText";
+import AttractionsIcon from "@mui/icons-material/Attractions";
+import SailingIcon from "@mui/icons-material/Sailing";
+import TakeoutDiningIcon from "@mui/icons-material/TakeoutDining";
+import { IconButton } from "@mui/material";
+import ExitToApp from "@mui/icons-material/ExitToApp";
+import { TextField } from "@mui/material";
+import { Typography } from "@mui/material";
+
+import MenuIcon from "@mui/icons-material/Menu";
+import HomeIcon from "@mui/icons-material/Home";
+import QuizIcon from "@mui/icons-material/Quiz";
+import CloseIcon from "@mui/icons-material/Close";
+import MenuBookIcon from "@mui/icons-material/MenuBook";
+import { Link } from "react-router-dom";
+
+import { onAuthStateChanged, signOut } from "firebase/auth";
+import { auth } from "../firebase";
+
+const linksData = ["", "quizzes", "quizzesAI", "onboarding", "guide"];
+
+export default function TemporaryDrawer({
+  logoutButton,
+  aiResponse,
+  clearAIResponse,
+  onDrawerOpen,
+  sendMessage,
+  handleAuthStateChanged,
+  handleLogout,
+  isLoggedIn,
+  loading,
+  historicalLandmarks,
+  natureParks,
+  politicalLandmarks,
+  setSelectedLandmarks,
+  renderMapComponent,
```

Comment:
I would advise to try and reduce the number of props on this component. This is quite a lot of props, and very hard to maintain when working with. Especially without any type checks etc, you will run into the problem of not knowing if those props are still used or if there are undefined props